test/parallel/test-http-agent-maxsockets.js

```diff
@@ -0,0 +1,50 @@
+var common = require('../common');
+var assert = require('assert');
+var http = require('http');
+var Agent = require('_http_agent').Agent;
+var EventEmitter = require('events').EventEmitter;
+
+var agent = new Agent({
+  keepAlive: true,
+  keepAliveMsecs: 1000,
+  maxSockets: 2,
+  maxFreeSockets: 2
+});
+
+var server = http.createServer(function (req, res) {
+  res.end('hello world');
+});
+
+function get(path, callback) {
+  return http.get({
+    host: 'localhost',
+    port: common.PORT,
+    agent: agent,
+    path: path
+  }, callback);
+}
+
+function done() {
+  var freepool = agent.freeSockets[Object.keys(agent.freeSockets)[0]];
+  assert.equal(freepool.length, 2, 'expect keep 2 free sockets, but got ' + freepool.length);
+  console.log('http keepalive agent maxsockets test success.');
+  process.exit(0);
```

Comment:
Please don't call process.exit() in tests, it tends to hide bugs.  The test should shut down naturally, i.e., because there are no more open sockets, timers, etc.